### PR TITLE
mcux-sdk: update to SDK 2.16.000

### PR DESCRIPTION
**Pull request by dleach02**
Updating mcux components to 2.16.100

Signed-off-by: David Leach <david.leach@nxp.com>

### CODE REVIEW

**Comment by lucien-nxp** [comment]
Rebase sdk_2.16.0 branch to newest commit on zephyr/main branch. Otherwise, it will case other platforms build failed on my RT1180 PRhttps://github.com/zephyrproject-rtos/zephyr/pull/73528

**Comment by lucien-nxp** [comment]
The building dependency under RT1189 don't be cherry-picked correctly, re-cherry-pick from original PR.

**Comment by lucien-nxp** [comment]
It seems that driver update involve some building errors on lvgl and Bluetooth cases on some platforms. 

272f84ca60afc759d81ed94100e78b37a859d3ff commit ID should be a appropriate base point, because zephyr's current track hal_nxp is this point, otherwise some S32K building issue will be introduced。
https://github.com/zephyrproject-rtos/zephyr/actions/runs/10195617084/job/28206275931

**Comment by lucien-nxp** [comment]
Hi @dleach02 ,
Why this commit is only selected hal_nxp.cmake file to new PR?
[mcux-sdk: add rt1180 build support](https://github.com/zephyrproject-rtos/hal_nxp/pull/391/commits/12520d78710c07116401ea16e6f0fae33b2f430f)

**Comment by dleach02** [comment]
> Hi @dleach02 , Why this commit is only selected hal_nxp.cmake file to new PR? [mcux-sdk: add rt1180 build support](https://github.com/zephyrproject-rtos/hal_nxp/pull/391/commits/12520d78710c07116401ea16e6f0fae33b2f430f) Could I cherry-pick directly on this PR?

@lucien-nxp please do cherry-pick what you need to this PR. I attempted to do that originally and missed things.

**Comment by lucien-nxp** [comment]
> > Hi @dleach02 , Why this commit is only selected hal_nxp.cmake file to new PR? [mcux-sdk: add rt1180 build support](https://github.com/zephyrproject-rtos/hal_nxp/pull/391/commits/12520d78710c07116401ea16e6f0fae33b2f430f) Could I cherry-pick directly on this PR?
> 
> @lucien-nxp please do cherry-pick what you need to this PR. I attempted to do that originally and missed things.

Thank you for quick respond, I have added an new commit which contain rt1180 building environment in this PR.

**Comment by lucien-nxp** [comment]
Hi @dleach02 ，
Why re-push and remove my newly added commit point?

**Comment by dleach02** [comment]
> Hi @dleach02 ， Why re-push and remove my newly added commit point?

This was probably a mistake on my part as I wasn't aware of a change. Please pull/check what I have and update with anything you believe is needed.

**Comment by lucien-nxp** [comment]
> > Hi @dleach02 ， Why re-push and remove my newly added commit point?
> 
> This was probably a mistake on my part as I wasn't aware of a change. Please pull/check what I have and update with anything you believe is needed.

Thank you, I have added my commit again on this PR, and rebase RT1180 basic enablement PR to newest to do test.

**Review by hakehuang — APPROVED** [review]
board regression test done, no issues found

**Comment by mmahadevan108** [comment]
Some new commits may have been added on Aug 8th : https://github.com/nxp-mcuxpresso/mcux-sdk/commits/main/

File: mcux/mcux-sdk/components/conn_fwloader/driver_conn_fwloader.cmake [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+# Copyright (c) 2023-24 NXP.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+#-------------------------------------------------------------------------------
+
 #Description: Driver conn_fwloader; user_visible: True
 include_guard(GLOBAL)
 message("driver_conn_fwloader component is included.")
 
+if(NOT CONFIG_BUILD_WITH_TFM)
```
Comment (mmahadevan108) [inline mcux/mcux-sdk/components/conn_fwloader/driver_conn_fwloader.cmake]: Have we submitted this change to SDK team

File: mcux/README
Comment (mmahadevan108) [inline mcux/README:1]: Looks like we have some merge stuff leftover on lines 7-11

File: mcux/mcux-sdk/drivers/mipi_dsi_split/fsl_mipi_dsi.c [hunk 2]
```diff
@@ -538,7 +538,7 @@ void DSI_SetDpiConfig(const MIPI_DSI_Type *base,
     assert(NULL != config);
 
     /* coefficient DPI event size to number of DSI bytes. */
-    float coff = ((float)numLanes * (float)dsiHsBitClkFreq_Hz) / ((float)dpiPixelClkFreq_Hz * 8.0);
+    float coff = ((float)numLanes * (float)dsiHsBitClkFreq_Hz) / ((float)dpiPixelClkFreq_Hz * 8.0f);
```
Comment (mmahadevan108) [inline mcux/mcux-sdk/drivers/mipi_dsi_split/fsl_mipi_dsi.c:541]: Have we submitted this change to the SDK team?

Comment (dleach02) [reply mcux/mcux-sdk/drivers/mipi_dsi_split/fsl_mipi_dsi.c:541]: I will submit all my patches as PRs to the SDK once we are clean and merged

File: mcux/mcux-sdk/components/conn_fwloader/driver_conn_fwloader.cmake (hunk 1 above)
Comment (dleach02) [reply mcux/mcux-sdk/components/conn_fwloader/driver_conn_fwloader.cmake]: will do once we have the full list

File: mcux/README
Comment (dleach02) [reply mcux/README:1]: addressed

**Review by JiafeiPan — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**